packages/e2e-tests/specs/performance/site-editor.test.js

```diff
@@ -75,112 +77,123 @@ describe( 'Site Editor Performance', () => {
 		}, html );
 		await saveDraft();
 
-		id = await page.evaluate( () =>
+		postId = await page.evaluate( () =>
 			new URL( document.location ).searchParams.get( 'post' )
 		);
 	} );
 
 	afterAll( async () => {
+		const resultsFilename = basename( __filename, '.js' ) + '.results.json';
+		writeFileSync(
+			join( __dirname, resultsFilename ),
+			JSON.stringify( results, null, 2 )
+		);
+
 		await deleteAllTemplates( 'wp_template' );
 		await deleteAllTemplates( 'wp_template_part' );
 		await activateTheme( 'twentytwentyone' );
 	} );
 
-	beforeEach( async () => {
-		await visitSiteEditor( {
-			postId: id,
-			postType: 'page',
-		} );
-	} );
-
-	it( 'Loading', async () => {
-		const editorURL = await page.url();
-
-		// Number of sample measurements to take.
+	describe( 'Loading', () => {
+		// Number of measurements to take.
 		const samples = 3;
 		// Number of throwaway measurements to perform before recording samples.
-		// Having at least one helps ensure that caching quirks don't manifest in
-		// the results.
+		// Having at least one helps ensure that caching quirks don't manifest
+		// in the results.
 		const throwaway = 1;
+		const iterations = Array.from(
+			{ length: samples + throwaway },
+			( _, i ) => i + 1
+		);
 
-		let i = throwaway + samples;
-
-		// Measuring loading time.
-		while ( i-- ) {
-			await page.close();
-			page = await browser.newPage();
+		it.each( iterations )(
+			`trace large post loading durations (%i of ${ iterations.length })`,
+			async ( i ) => {
+				// Open the test page in Site Editor.
+				await visitSiteEditor( {
+					postId,
+					postType: 'page',
+					path: '/navigation/single',
+				} );
+
+				// Wait for the canvas to become editable.
+				await canvas().waitForSelector(
+					'[data-rich-text-placeholder="Write site tagline…"]'
+				);
```

Comment:
In my testing, this reliably detects when the page content is ready to be edited. No need to wait for other selectors. https://github.com/WordPress/gutenberg/blob/d57b42970fa7540b1b46ad92cde45b3dbcab5823/packages/block-library/src/site-tagline/edit.js#L55-L60

Reply:
I think `canUserEdit` here indicates whether the current user has a role to update the site tagline, not whether the content is editable.
What about using a `[data-type="core/post-content"] [data-type="core/paragraph"]` selector, as in the typing test?

Reply:
> What about using a [data-type="core/post-content"] [data-type="core/paragraph"] selector, as in the typing test?

See https://github.com/WordPress/gutenberg/pull/48138#discussion_r1113032311 - it's for the same reason. If it's possible to fix the content flashing, the generic paragraph selector should become sufficient.